### PR TITLE
Azure: Allow to manage machine with identity

### PR DESCRIPTION
**Pull request by dkistner**
Allow to create and manage Azure virtual machines which have an user assigned managed identity assigned to.

**What this PR does / why we need it**:
Azure machines can now be deployed with an attached user assigned managed identity, which can be used to authenticate to other Azure services like ACR.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement user
Azure machines can now be deployed with an attached user assigned managed identity.
```

### CODE REVIEW

File: pkg/driver/driver_azure.go [hunk 1]
```diff
@@ -163,6 +163,15 @@ func (d *AzureDriver) getVMParameters(vmName string, networkInterfaceReferenceID
 		}
 	}
 
+	if d.AzureMachineClass.Spec.Properties.IdentityID != nil {
```
Comment (ggaurav10) [inline pkg/driver/driver_azure.go]: ` && *d.AzureMachineClass.Spec.Properties.IdentityID != ""`?

Comment (dkistner) [reply pkg/driver/driver_azure.go]: Thanks I have adapted this.

**Comment by dkistner** [comment]
Thanks @ggaurav10. Should be done.

File: pkg/openapi/openapi_generated.go [hunk 2]
```diff
@@ -9834,7 +9798,7 @@ func GetOpenAPIDefinitions(ref common.ReferenceCallback) map[string]common.OpenA
 					Properties: map[string]spec.Schema{
 						"replicas": {
 							SchemaProps: spec.SchemaProps{
-								Description: "Replicas is the number of desired replicas. This is a pointer to distinguish between explicit zero and unspecified. Defaults to 0. More info: https://kubernetes.io/docs/concepts/workloads/controllers/replicationcontroller#what-is-a-replicationcontroller",
+								Description: "Replicas is the number of desired replicas. This is a pointer to distinguish between explicit zero and unspecified. Defaults to 1. More info: https://kubernetes.io/docs/concepts/workloads/controllers/replicationcontroller#what-is-a-replicationcontroller",
```
Comment (ggaurav10) [inline pkg/openapi/openapi_generated.go]: This shouldn't change with this PR. Is it a rebasing issue?

**Comment by dkistner** [comment]
Hmm I just called `hack/generate-code` and this change was produced by the script. I have build this based on the master. Is there something which needs to be fixed on the upstream master? If yes give me a hint when I need to rebase and re run the code generation.

**Comment by hardikdr** [comment]
Sure @dkistner, We haven't yet been able to look into it. We shall soon fix it and let you know.

**Comment by hardikdr** [comment]
One of the recent PR dint run `./hack/generatecode`, which brought these changes into this PR.
We can merge this one.

File: pkg/openapi/openapi_generated.go (hunk 2 above)
Comment (hardikdr) [reply pkg/openapi/openapi_generated.go]: This got in as `hack/generatecode` was not run previously. Though we need to check on the open-api generate-code path, as in reality it gets defaulted to `0`, but not the concern of this PR.

**Pull request merged**